File: docs/00-Foundations-and-Complexity/js/Right_angle_pattern.js
// Run using: node Right_angle_pattern.js [size]
const args = process.argv.slice(2);
const n = args.length > 0 ? parseInt(args[0], 10) : 5;

if (args.length === 0) {
    console.log("Usage: node Right_angle_pattern.js [size]");
    console.log("Defaulting to size 5:\n");
}

for (let i = 1; i <= n; i++) {
    console.log('*'.repeat(i));
}

File: docs/00-Foundations-and-Complexity/cpp/Right_angle_pattern.cpp
#include <iostream>
#include <string>

int main(int argc, char* argv[]) {
    int n = 5;
    if (argc > 1) {
        try {
            n = std::stoi(argv[1]);
        } catch (...) {
            std::cout << "Invalid argument. Defaulting to 5.\n";
        }
    } else {
        std::cout << "Usage: ./Right_angle_pattern [size]\n";
        std::cout << "Defaulting to size 5:\n\n";
    }
    
    for (int i = 1; i <= n; ++i) {
        std::cout << std::string(i, '*') << "\n";
    }
    return 0;
}
